heron/scheduler-core/tests/java/com/twitter/heron/scheduler/SubmitterMainTest.java

```diff
@@ -181,8 +183,31 @@ public void testSubmitTopologyLauncherException() throws Exception {
     submitterMain.submitTopology();
   }
 
+  @Test(expected = SubmitDryRunResponse.class)
+  public void testSubmitTopologyDryRun() throws Exception {
+    SchedulerStateManagerAdaptor adaptor = mock(SchedulerStateManagerAdaptor.class);
+    PowerMockito.whenNew(SchedulerStateManagerAdaptor.class).withAnyArguments().
+        thenReturn(adaptor);
+    SubmitterMain submitterMain = spy(new SubmitterMain(config, topology));
+    PowerMockito.mockStatic(Context.class);
+    PowerMockito.when(Context.dryRun(config)).thenReturn(true);
+    doThrow(new SubmitDryRunResponse(topology, config, mock(PackingPlan.class))).
```

Comment:
resolved. see: https://github.com/twitter/heron/pull/1629/files#diff-9d7c5b93dadaa33d2766bdfdad746d08

minor comment: i have to mock the behavior of `Context` instead of `config` object since mocking `config` does not work... (NPE)